forc/src/cli/commands/plugins.rs

```diff
@@ -18,15 +22,62 @@ pub struct Command {
     describe: bool,
 }
 
+fn get_file_name(path: &Path) -> String {
+    if let Some(Some(path_str)) = path.file_name().map(|path_str| path_str.to_str()) {
+        path_str.to_owned()
+    } else {
+        path.display().to_string()
+    }
+}
+
 pub(crate) fn exec(command: PluginsCommand) -> ForcResult<()> {
     let PluginsCommand {
         print_full_path,
         describe,
     } = command;
 
+    let mut plugins = crate::cli::plugin::find_all()
+        .map(|path| print_plugin(path.clone(), print_full_path, describe).map(|info| (path, info)))
+        .collect::<Result<Vec<(_, _)>, _>>()?
+        .into_iter()
+        .fold(HashMap::new(), |mut acc, (path, content)| {
+            let bin_name = get_file_name(&path);
+            acc.entry(bin_name.clone())
+                .or_insert_with(|| (bin_name, vec![], content.clone()))
+                .1
+                .push(path);
+            acc
+        })
+        .into_values()
+        .map(|(bin_name, paths, content)| {
+            let mut temp_hashmap = HashMap::new();
+            (
+                bin_name,
+                paths
+                    .into_iter()
+                    .filter_map(|path| {
+                        if temp_hashmap.get(&path).is_some() {
+                            return None;
+                        }
+                        temp_hashmap.insert(path.clone(), true);
+                        Some(path)
```

Comment:
I am a little lost about what is happening here,

After this step do we have bin name tupled with path for the first occurance and others are tupled with None? Is that even neccesary? We are not removing them from the vec so that we cna check the len in line 75. So why are we making the duplicate entries `None` explicitly.